app/src/androidTest/java/com/google/maps/android/compose/MapInColumnTests.kt

```diff
@@ -0,0 +1,141 @@
+        val countDownLatch = CountDownLatch(1)
+        composeTestRule.setContent {
+            var scrollingEnabled by remember { mutableStateOf(true) }
+
+            LaunchedEffect(cameraPositionState.isMoving) {
+                if (!cameraPositionState.isMoving) {
+                    scrollingEnabled = true
+                    Log.d(TAG, "Map camera stopped moving - Enabling column scrolling...")
+                }
+            }
+
+            MapInColumn(
+                modifier = Modifier.fillMaxSize(),
+                cameraPositionState,
+                columnScrollingEnabled = scrollingEnabled,
+                onMapTouched = {
+                    scrollingEnabled = false
+                    Log.d(
+                        TAG,
+                        "User touched map - Disabling column scrolling after user touched this Box..."
+                    )
+                },
+                onMapLoaded = {
+                    countDownLatch.countDown()
+                }
+            )
+        }
+        val mapLoaded = countDownLatch.await(30, TimeUnit.SECONDS)
+        assertTrue("Map loaded", mapLoaded)
+    }
+
+    @Before
+    fun setUp() {
+        cameraPositionState = CameraPositionState(
+            position = CameraPosition.fromLatLngZoom(
+                startingPosition,
+                startingZoom
+            )
+        )
+    }
+
+    @Test
+    fun testStartingCameraPosition() {
+        initMap()
+        startingPosition.assertEquals(cameraPositionState.position.target)
+    }
+
+    @Test
+    fun testLatLngInVisibleRegion() {
+        initMap()
+        composeTestRule.runOnUiThread {
+            val projection = cameraPositionState.projection
+            assertNotNull(projection)
+            assertTrue(
+                projection!!.visibleRegion.latLngBounds.contains(startingPosition)
+            )
+        }
+    }
+
+    @Test
+    fun testLatLngNotInVisibleRegion() {
+        initMap()
+        composeTestRule.runOnUiThread {
+            val projection = cameraPositionState.projection
+            assertNotNull(projection)
+            val latLng = LatLng(23.4, 25.6)
+            assertFalse(
+                projection!!.visibleRegion.latLngBounds.contains(latLng)
+            )
+        }
+    }
+
+    @Test
+    fun testScrollColumn_MapCameraRemainsSame() {
+        initMap()
+        // Check that the column scrolls to the last item
+        composeTestRule.onRoot().performTouchInput { swipeUp() }
+        composeTestRule.waitForIdle()
+        composeTestRule.onNodeWithTag("Item 1").assertIsNotDisplayed()
+
+        // Check that the map didn't change
+        startingPosition.assertEquals(cameraPositionState.position.target)
+    }
+
+//    @Test
+//    fun testPanMapUp_MapCameraChangesColumnDoesNotScroll() {
+//        initMap()
+//        // Swipe the map up
+//        // FIXME - for some reason this scrolls the entire column instead of just the map
```

Comment:
@arriolac @wangela I still need to work on this unit test further (for some reason it fails when it should pass), but I'd like to go ahead and get this PR merged first and add this test later (IMHO it's a "nice-to-have"). Otherwise, due to the refactoring in this PR I keep getting merge conflicts every time the main branch is changed.

So it should be ready for review.